### PR TITLE
Agent docs improvements

### PR DESCRIPTION
**Pull request by hassanshaikley**
Hey all,

Here are some humble improvements to `Agent` docs imo.

Let me know if you have any suggestions! Or anything else in the Agent docs you would like for me to look at.

### CODE REVIEW

File: lib/elixir/lib/agent.ex [hunk 1]
```diff
@@ -74,8 +73,8 @@ defmodule Agent do
   at least initially, or small enough to be sent to the client cheaply. Another
   factor is whether the data needs to be processed atomically: getting the
   state and calling `do_something_expensive(state)` outside of the agent means
-  that the agent's state can be updated in the meantime. This is specially
-  important in case of updates as computing the new state in the client rather
+  that the agent's state can be updated in the meantime. This is especially
```
Comment (josevalim) [inline lib/elixir/lib/agent.ex]: I am not a native speaker but I believe "specially" works best here? "specially" is about being distinctive, "especially" is about being noteworthy.

File: lib/elixir/lib/agent.ex [hunk 2]
```diff
@@ -74,8 +73,8 @@ defmodule Agent do
...
+  important when updating the state as computing the new state in the client rather
```
Comment (josevalim) [inline lib/elixir/lib/agent.ex]: I am not sure if the double `-ing` reads well here: "... updating ... computing ...".

File: lib/elixir/lib/agent.ex (hunk 1 above)
Comment (hassanshaikley) [reply lib/elixir/lib/agent.ex]: I had no idea it was an actual word! TIL.

My bad!

File: lib/elixir/lib/agent.ex (hunk 2 above)
Comment (hassanshaikley) [reply lib/elixir/lib/agent.ex]: Hmm I find it easier to make sense of in spite of the double `-ing` personally. But I could see how others would not!

File: lib/elixir/lib/agent.ex [hunk 3]
```diff
@@ -21,7 +21,7 @@ defmodule Agent do
         end
 
         def value do
-          Agent.get(__MODULE__, & &1)
+          Agent.get(__MODULE__, &(&1))
```
Comment (hassanshaikley) [inline lib/elixir/lib/agent.ex]: Also how do you feel about that. It seems inconsistent that parens aren't always used.

Comment (hassanshaikley) [reply lib/elixir/lib/agent.ex]: nevermind I think this is a pattern!

**Comment by hassanshaikley** [comment]
I understand if y'all want to close this, I have no hard feelings! 

These are just some improvements I thought might be nice. 😄 

💜

File: lib/elixir/lib/agent.ex [hunk 4]
```diff
@@ -136,7 +135,7 @@ defmodule Agent do
 
   ## Name registration
 
-  An agent is bound to the same name registration rules as GenServers.
+  An agent is bound to the same name registration rules as `GenServer`s.
```
Comment (josevalim) [inline lib/elixir/lib/agent.ex]: We already link to GenServer next, so the current approach is fine. :)

**Comment by josevalim** [comment]
Hi @hassanshaikley, thanks for the PR. Since those are minor changes that we are not super sure about, let's keep things as is, thanks!

**Comment by hassanshaikley** [comment]
Of course! 😄

Have an excellent night!

**Pull request closed without merging**